File: java/src/main/java/org/shypl/sna/impl/VkSocialNetwork.java
package org.shypl.sna.impl;

import org.shypl.common.util.DeclinationImpl;
import org.shypl.sna.Currency;
import org.shypl.sna.SocialNetwork;

public final class VkSocialNetwork extends SocialNetwork {
	public VkSocialNetwork() {
		super(1, "vk", "ВКонтакте", new Currency("голоса", new DeclinationImpl("голос", "голоса", "голосов")));
	}
}

File: php/src/org/shypl/sna/impl/VkSocialNetwork.php
<?php

namespace org\shypl\sna\impl;

use org\shypl\sna\Adapter;
use org\shypl\sna\SocialNetwork;

final class VkSocialNetwork extends SocialNetwork {
	const ID = 1;

	/**
	 *
	 */
	public function __construct() {
		parent::__construct(self::ID, "vk");
	}

	/**
	 * @param array $parameters
	 *
	 * @return Adapter
	 */
	public function createAdapter(array $parameters) {
		return new VkAdapter($parameters);
	}
}